core/src/main/scala/gnieh/diffson/JsonPointerSupport.scala

```diff
@@ -18,34 +18,67 @@ package gnieh.diffson
 
 import scala.annotation.tailrec
 
+import scala.language.implicitConversions
+
 trait JsonPointerSupport[JsValue] {
   this: DiffsonInstance[JsValue] =>
 
   import provider._
 
-  implicit val pointer = new JsonPointer()
-
   /** A class to work with Json pointers according to http://tools.ietf.org/html/rfc6901.
    *  The behavior in case of invalid pointer is customizable by passing an error handler
    *  when instantiating.
    *
    *  @author Lucas Satabin
    */
-  class JsonPointer {
+  case class JsonPointer(path: Pointer) {
+
+    /** Evaluates the given path in the given JSON object.
+     *  Upon missing elements in value, the error handler is called with the current value and element
+     */
+    final def evaluate(value: JsValue): JsValue = evaluate(value, this, Pointer.Root)
```

Comment:
The problem is that here it cannot be overridden modularly, because when compiling the only one available is the default handler, and it cannot be changed when using diffson. I would remove the one in `JsonProvider` and use a specific `LowPriorityImplicits` trait mixed in with a `DiffsonInstance`. This should solve the problem while allowing for overriding at call site.